packages/core/src/WorkflowExecute.ts

```diff
@@ -199,10 +199,14 @@ export class WorkflowExecute {
 						if (node && pinData && pinData[node.name]) {
 							incomingData.push(pinData[node.name]);
 						} else {
-							incomingData.push(
-								// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-								runData[connection.node][runIndex].data![connection.type][connection.index]!,
-							);
+							if (
+								runData[connection.node][runIndex] &&
+								runData[connection.node][runIndex].data![connection.type][connection.index]
+							) {
+								incomingData.push(
+									runData[connection.node][runIndex].data![connection.type][connection.index]!,
+								);
+							}
```

Comment:
I find repetitive code like this harder to read and debug. 
How about something like this instead?
```suggestion
							const nodeIncomingData =
								runData[connection.node][runIndex]?.data?.[connection.type]?.[connection.index];
							if (nodeIncomingData) {
								incomingData.push(nodeIncomingData);
							}
```

Reply:
Thanks, I love the suggestion. I'll run some more tests locally and then merge once e2e tests pass